### PR TITLE
Fixed the USB logger

### PR DESCRIPTION
**Pull request by rmeertens**
The USB logger works again! Made it into the current setup where you can specify which camera to log.

### CODE REVIEW

**Comment by gautierhattenberger** [comment]
it looks fine but I can't really test it

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 1]
```diff
@@ -28,35 +28,118 @@
...
+#include <sys/stat.h>
+#include <unistd.h>
+
 
 /** Set the default File logger path to the USB drive */
 #ifndef VIDEO_USB_LOGGER_PATH
 #define VIDEO_USB_LOGGER_PATH /data/video/usb
 #endif
 
+#ifndef VIDEO_LOG_WIDTH
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: please document newly added define options in the module xml

File: conf/modules/video_usb_logger.xml [hunk 2]
```diff
@@ -7,12 +7,14 @@
...
   </doc>
   <depends>video_rtp_stream</depends>
   <header>
     <file name="video_usb_logger.h"/>
   </header>
-  <periodic fun="video_usb_logger_periodic()" start="video_usb_logger_start()" stop="video_usb_logger_stop()" autorun="FALSE"/>
+  
+  <periodic fun="video_usb_logger_periodic()" start="video_usb_logger_start()" stop="video_usb_logger_stop()" autorun="TRUE"/>
   <makefile target="ap">
     <file name="video_usb_logger.c"/>
```
Comment (flixr) [inline conf/modules/video_usb_logger.xml]: please remove unused defines....

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 3]
```diff
@@ -28,35 +28,118 @@
...
+  sprintf(save_name, "%s/img_%05d.jpg", foldername, shotNumber);
+
+  shotNumber++;
+  // Check if file exists or not
+  if (access(save_name, F_OK) == -1) {
+
+    // Create a high quality image (99% JPEG encoded)
+    jpeg_encode_image(img, img_jpeg, 99, TRUE);
+
+#if JPEG_WITH_EXIF_HEADER
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: this is an option that you should define in your module/airframe file?
Maybe it should be specific to this module?

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 4]
```diff
@@ -28,35 +28,118 @@
...
+    counter++;
+  }
+
+}
+
+struct image_t *log_image(struct image_t *img);
+struct image_t *log_image(struct image_t *img)
+{
+  save_shot(img, &img_jpeg);
+  return FALSE;
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: should probably return `img`

**Comment by rmeertens** [comment]
Added all documentation!

**Comment by flixr** [comment]
Thx, but I still think it would be good to use a common prefix for the defines, right now there are
- VIDEO_USB_LOGGER_
- VIDEO_USBLOG_
- VIDEO_LOG_
- and no prefix for JPEG_WITH_EXIF_HEADER

**Comment by flixr** [comment]
also why is it named udb_logger in the first place? It has nothing to do with USB...

**Comment by rmeertens** [comment]
Changed the names of the defines.
Added Christophe as author before my changes (I think it is important to see who worked on it in case of questions)
For now, let's keep it at USB logger. Originally the path was set the the USB stick of the ARDrone.

**Comment by flixr** [comment]
thx!

**Pull request merged**